kratos/utilities/sparse_matrix_multiplication_utility.h

```diff
@@ -683,6 +683,267 @@ class SparseMatrixMultiplicationUtility
         }
     }
 
+    /**
+     * @brief This method assembles several sparse matrices into one large sparse matrix
+     * @param rMatricespBlocks The pointers to the matrices we are interested in assemble
+     * @param ContributionCoefficients The matrix containing the coefficients to be considered (copy, so we don't need to provide it)
+     * @param TransposeBlocks The matrix containing the flags telling us to transpose the blocks (copy, so we don't need to provide it)
+     */
+    template <class TMatrix>
```

Comment:
are u sure you want to template in TMatrix? anyway to do it efficiently you need to use the internals of the Matrix definition, so i don't think this makes sense

Reply:
Basically I am following the design considered in the rest of the sparse_matrix_mutiplication_utilities